### PR TITLE
Fix LIMIT OFFSET for BigQuery DIalect

### PR DESCRIPTION
**Pull request by goldmedal**

### CODE REVIEW

File: trino-parser/src/main/java/io/trino/sql/SqlFormatter.java [hunk 1]
```diff
@@ -378,8 +379,14 @@ protected Void visitQuery(Query node, Integer indent)
...
-            node.getOffset().ifPresent(offset -> process(offset, indent));
-            node.getLimit().ifPresent(limit -> process(limit, indent));
+            if (dialect.equals(BIGQUERY)) {
+                node.getLimit().ifPresent(limit -> process(limit, indent));
+                node.getOffset().ifPresent(offset -> process(offset, indent));
+            }
+            else {
+                node.getOffset().ifPresent(offset -> process(offset, indent));
+                node.getLimit().ifPresent(limit -> process(limit, indent));
+            }
```
Comment (brandboat) [inline trino-parser/src/main/java/io/trino/sql/SqlFormatter.java:389]: Does the order matters ? Looks like both if-else blocks are almost doing the same thing.

File: trino-parser/src/main/java/io/trino/sql/SqlFormatter.java [hunk 2]
```diff
@@ -414,8 +421,14 @@ protected Void visitQuerySpecification(QuerySpecification node, Integer indent)
...
```
Comment (brandboat) [inline trino-parser/src/main/java/io/trino/sql/SqlFormatter.java:431]: Same thing here.

File: accio-main/src/main/java/io/accio/main/connector/bigquery/BigQuerySqlConverter.java [hunk 3]
```diff
@@ -81,11 +81,14 @@ public String convert(@Language("sql") String sql, SessionContext sessionContext
         LOG.info("[Input sql]: %s", sql);
 
         for (SqlRewrite rewrite : sqlRewrites) {
-            LOG.debug("Before %s: %s", rewrite.getClass().getSimpleName(), formatSql(rewrittenNode));
+            if (LOG.isDebugEnabled()) {
```
[the remark this answers is not in the document]
Comment (goldmedal) [reply accio-main/src/main/java/io/accio/main/connector/bigquery/BigQuerySqlConverter.java:84]: IMO, If we don't add this condition, `formatSql` will be invoked but won't be show.

File: trino-parser/src/main/java/io/trino/sql/SqlFormatter.java (hunk 1 above)
Comment (goldmedal) [reply trino-parser/src/main/java/io/trino/sql/SqlFormatter.java:389]: Yes, it's an ordering problem. BigQuery need to place `OFFSET` behind `LIMIT`.

**Review by brandboat — APPROVED** [review]
lgtm

**Pull request merged**